src/PhpSlang/Collection/Generic/Component/IterableCollection.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace PhpSlang\Collection\Generic\Component;
+
+trait IterableCollection
+{
+    use CollectionWithContent;
+
+    public function rewind()
+    {
+        reset($this->content);
```

Comment:
I'm not quite sure if it's a good solution. Maybe we should create a new instance to make it immutable...